File: src/test/java/nl/mvdr/adventofcode/adventofcode2024/day16/Part1Test.java
package nl.mvdr.adventofcode.adventofcode2024.day16;

import java.util.stream.Stream;

import org.junit.jupiter.params.provider.Arguments;

import nl.mvdr.adventofcode.solver.SolverTest;

public class Part1Test extends SolverTest<Part1> {

    public Part1Test() {
        super(Part1.class);
    }
    
    static Stream<Arguments> testSolution() {
        return Stream.of(
                Arguments.of("7036", "example-day16-2024-0.txt"),
                Arguments.of("11048", "example-day16-2024-1.txt"));
    }
}

File: adventofcode-2025/src/test/kotlin/nl/mvdr/adventofcode/adventofcode2025/day03/Part1Test.kt
package nl.mvdr.adventofcode.adventofcode2025.day03

import nl.mvdr.adventofcode.solver.FunctionSolverTest
import org.junit.jupiter.params.provider.Arguments

class Part1Test: FunctionSolverTest<Int>(::solvePart1) {
    companion object {
        @JvmStatic
        fun testSolution(): List<Arguments> {
            return listOf(
                Arguments.of("?", "example-day03.txt")
            )
        }
    }
}